src/components/RadioButtonGroup/index.js

```diff
@@ -45,8 +61,62 @@ class RadioButtonGroup extends Component {
         return undefined;
     }
 
+    getCheckedOptionRef() {
+        const { value, options } = this.props;
+        const currentOptionIndex = options.findIndex(option => isOptionChecked(option, value));
+        return currentOptionIndex !== -1 ? this.optionsRefs[currentOptionIndex] : null;
+    }
+
+    generateRefsForOptions() {
+        const { options } = this.props;
+        return options.reduce(list => [...list, React.createRef()], []);
+    }
+
+    addRefsToOptions(options) {
+        return options.reduce(
+            (list, option, index) => [
+                ...list,
+                {
+                    optionRef: this.optionsRefs[index],
+                    ...option,
+                },
+            ],
+            [],
+        );
+    }
```

Comment:
here you can use:
```
return options.map(option => ({
            ...option,
            optionRef: React.createRef(),
        }));
```

Reply:
the point of this function is to assign a previusly created reference to an option... createRef here will just create a new reference that we do not know.... the idea of this is that you have to different lists one with the options and one with the references...  remember that when value change we have no control over it.... and the value component receive is a value without ref... so we have to search for that value on the original options list and then use it index to get the ref on the refs list... so if we use this code you suggest... we are just reassigning a new useless ref to the options list ..

Reply:
You are right but I mean not to use reduce